src/components/Common/UI/Switch/Switch.tsx

```diff
@@ -0,0 +1,68 @@
+import { Switch as _Switch, type SwitchProps as AriaSwitchProps } from 'react-aria-components'
+import { useFieldIds } from '../hooks/useFieldIds'
+import {
+  HorizontalFieldLayout,
+  type SharedHorizontalFieldLayoutProps,
+} from '../HorizontalFieldLayout'
+import styles from './Switch.module.scss'
+
+export interface SwitchProps
+  extends SharedHorizontalFieldLayoutProps,
+    Omit<AriaSwitchProps, 'children'> {
```

Comment:
We probs don't want to extend AriaSwitchProps since we're trying to keep our interfaces completely defined on our end. If we do keep this IMO we should specifically use Pick to pull out what we need. my hope would be we could just reference native html props to supplement where needed instead of theirs tho